File: Python/75_sort_color.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        b, e, i = 0,len(nums)-1, 0
        while i <= e:
            while nums[i] == 2 and i < e:
                nums[i], nums[e] = nums[e], nums[i]
                e -= 1
            while nums[i] == 0 and i > b:
                nums[i], nums[b] = nums[b], nums[i]
                b += 1
            i += 1

File: golang/75_sort_color.go
package main

func sortColors(nums []int)  {
	b := 0
	e := len(nums) - 1
	cur := 0
	for cur <= e{
		for nums[cur] == 2 && cur > b{
			tmp := nums[cur]
			nums[cur] = nums[b]
			nums[b] = tmp
			b += 1
		}
		for nums[cur] == 0 && cur < e{
			tmp := nums[cur]
			nums[cur] = nums[e]
			nums[e] = tmp
			e -= 1
		}
		cur += 1
	}
}

